### DISCUSSION TITLE
preventsLazyLoading only works with multiple records?

### DISCUSSION THREAD

**Discussion by frits-metalogix** [laravel/framework#45328] [Q&A]
It looks like preventsLazyLoading is only enforced for multiple records? I've tried enforcing it in a test with only one record entered into the database, my colleague had an issue related to this in our QA environment when he was using an API, he had multiple records. I don't think I've come accross this in the docs, or maybe this can be removed, the DX felt a bit weird? Hope I don't come off as winy.

https://github.com/laravel/framework/blob/442626c5c9a01c7be6b627fde07081c117645831/src/Illuminate/Database/Eloquent/Builder.php#L399

**Comment by joligoms** [comment]
This is also really bothering me. I don't get it :(

**Comment by Angel5a** [comment]
Yeah. Having this count check is really hinders in tests.

This flag is intended to solve N+1 problem and only this problem. Someone decided, that 1+1 would be 1+1 in no mater, would you eager load relations, or lazy load them, and it isn't a problem. And this is a drawback of it's name preventsLazyLoading and not a solveNPlusOne, and it doesn't really helps in solving that problem. But it's better then nothing. You just have to remember to always create a bunch of objects.
